pkg/manifest/presentation.go

```diff
@@ -0,0 +1,120 @@
+package manifest
+
+import "encoding/json"
+
+/**
+ * The Presentation Hints extension defines a number of hints for User Agents about the way content
+ * should be presented to the user.
+ *
+ * https://readium.org/webpub-manifest/modules/presentation.html
+ * https://readium.org/webpub-manifest/schema/extensions/presentation/metadata.schema.json
+ *
+ * These properties are nullable to avoid having default values when it doesn't make sense for a
+ * given [Publication]. If a navigator needs a default value when not specified,
+ * Presentation.DEFAULT_X and Presentation.X.DEFAULT can be used.
+ */
+type Presentation struct {
+	Clipped     *bool        `json:"clipped,omitempty"`     // Specifies whether or not the parts of a linked resource that flow out of the viewport are clipped.
+	Continuous  *bool        `json:"continuous,omitempty"`  // Indicates how the progression between resources from the [readingOrder] should be handled.
+	Fit         *Fit         `json:"fit,omitempty"`         // Suggested method for constraining a resource inside the viewport.
+	Orientation *Orientation `json:"orientation,omitempty"` // Suggested orientation for the device when displaying the linked resource.
+	Overflow    *Overflow    `json:"overflow,omitempty"`    // Suggested method for handling overflow while displaying the linked resource.
+	Spread      *Spread      `json:"spread,omitempty"`      // Indicates the condition to be met for the linked resource to be rendered within a synthetic spread.
+	Layout      *EpubLayout  `json:"layout,omitempty"`      // Hints how the layout of the resource should be presented (EPUB extension).
+}
+
+const PRESENTATION_DEFAULT_CLIPPED = false    // Default value for Presentation.Clipped
+const PRESENTATION_DEFAULT_CONTINUOUS = false // Default value for Presentation.Continuous
+
+type Fit string // Suggested method for constraining a resource inside the viewport.
+const (
+	FIT_WIDTH   Fit = "width"
+	FIT_HEIGTH  Fit = "height"
+	FIT_CONTAIN Fit = "contain"
+	FIT_COVER   Fit = "cover"
+)
```

Comment:
The convention in Go is to use `MixedCase` instead of `UPPERCASE_UNDERSCORE`, even for constants.

See https://golang.org/doc/effective_go#mixed-caps

Reply:
Done